lib/rules/no-unused-expressions.js

```diff
@@ -211,7 +216,8 @@ module.exports = {
 			ExpressionStatement(node) {
 				if (
 					Checker.isDisallowed(node.expression) &&
-					!isDirective(node)
+					!astUtils.isDirective(node) &&
+					!(ignoreDirectives && isDirective(node))
```

Comment:
Can you add a test where `ignoreDirectives` is `true` and the rule reports something that isn't a directive? To verify https://github.com/eslint/eslint/pull/19645#discussion_r2053459468 is resolved and ensure `isDirective()` is called.